lib/Driver/ToolChains.cpp

```diff
@@ -1287,6 +1287,26 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   if (!Linker.empty()) {
     Arguments.push_back(context.Args.MakeArgString("-fuse-ld=" + Linker));
   }
+  
+  // Configure the toolchain
+  // By default, use the system clang++ to link.
+  StringRef Clang = "clang++";
+  if (const Arg *A = context.Args.getLastArg(options::OPT_toolchain)) {
```

Comment:
Yes, it works in that case (that's the exact case I'm doing at home). It doesn't do anything if you compile for a Darwin target though - swift will directly call 'ld' and not look for a linker inside the toolchain folder. That doesn't matter so much because you can only compile for Darwin from OSX, and the OSX linker is capable of cross-linking for other Darwin targets.

Are you saying perhaps that we should also look for ld in the toolchain folder for Darwin targets? Would give consistency and possibly help linker development.